CanHazFunny/CanHazFunny/Jester.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace CanHazFunny
+{
+    public class Jester
+    {
+        private IJokeService _JokeService;
+        private IJokeDisplay _DisplayService;
+
+        public IJokeService JokeService { get => _JokeService; }
+
+        public IJokeDisplay DisplayService { get => _DisplayService; }
+
+        public Jester(IJokeService jokeService, IJokeDisplay tellerService)
+        {
+            _JokeService = jokeService ?? throw new System.ArgumentNullException(nameof(jokeService));
+
+            _DisplayService = tellerService ?? throw new System.ArgumentNullException(nameof(tellerService));
+        }
+
+        public void TellJoke()
+        {
+            string joke = JokeService.GetJoke();
+
+            while(joke.Contains("Chuck") | joke.Contains("Norris"))

```

Comment:
Though this technically works, there is a subtle difference between `|` and `||`. The first is a bitwise or, the second is a logical or. Unless you are looking for bit-wise operations, most of the time you want to use `||`
```suggestion
            while(joke.Contains("Chuck") || joke.Contains("Norris"))

```